03_python_fastapi_project/main.py

```
@@ -111,6 +112,126 @@ async def delete_product(product_id: int, db: AsyncSession = Depends(get_db)):
+    """
+    Get all items in the basket.
+    """
+    result = await db.execute(
+        select(BasketItem).options(selectinload(BasketItem.product))
+    )
+    basket_items = result.scalars().all()
+    return basket_items
+
+
+@app.post("/basket/", response_model=BasketItemDTO, status_code=status.HTTP_201_CREATED)
+async def add_to_basket(item: BasketItemCreate, db: AsyncSession = Depends(get_db)):
```

Ruff (0.13.3)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)